Sources/Core/Model/Attachment.swift

```diff
@@ -211,9 +212,73 @@ public extension Attachment {
 }
 
 /// An attachment type.
-public enum AttachmentType: String, Codable {
+public enum AttachmentType: RawRepresentable, Codable, Equatable {
     /// An attachment type.
-    case unknown, image, imgur, giphy, video, youtube, product, file, link
+    case unknown(raw: String?)
```

Comment:
The `unknown` type should be unknown.
Here needs to add custom type and not optional string parameter:
```
case unknown
case custom(type: String)
```
Please make all changes that needed for that.

Reply:
I was not sure what to do about this one, I saw on API docs somewhere that `type` can't be null. I'm not sure if `unknown` will ever happen but I changed it according to your suggestion :)

Reply:
The `unknown` type is for decoding only.